### PR TITLE
(DOCSP-26537): Updating list of Kotlin-supported data types

### PR DESCRIPTION
**Pull request by cbullinger**
### Jira

- https://jira.mongodb.org/browse/DOCSP-26537

### Staged Changes

- [Supported Types - Kotlin SDK](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/docsp-26537-kotlin-primitives-list/sdk/kotlin/realm-database/schemas/supported-types/)

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/kotlin/realm-database/schemas/supported-types.txt [hunk 1]
```diff
@@ -17,15 +17,26 @@ Realm Database supports the following field data types:
...
 - ``Double``
 - ``Float``
-- ``ObjectId``
+- ``ObjectId`` 
 - ``RealmInstant``
 - Any ``RealmObject`` subclass
-- ``RealmList``
+- ``RealmList<T>``, where T is any of the supported data types or a   
+  `RealmObject <{+kotlin-local-prefix+}io.realm.kotlin.types/-realm-object/index.html>`__.
+  Unless T is a ``RealmObject``, T can also be nullable (``RealmList<T?>``).
```
Comment (mongomoe) [inline source/sdk/kotlin/realm-database/schemas/supported-types.txt:32]: Suggestion, I think the following would read better (but you can take this advice or leave it):

```suggestion
   Lists of ``RealmObject`` cannot have null elements. All other types of ``RealmList<T>`` can be nullable (``RealmList<T?>``).
```

File: source/sdk/kotlin/realm-database/schemas/supported-types.txt [hunk 2]
```diff
@@ -17,15 +17,26 @@ Realm Database supports the following field data types:
...
+- ``RealmSet<T>``, where T is any of the supported data types or a
+  `RealmObject <{+kotlin-local-prefix+}io.realm.kotlin.types/-realm-object/index.html>`__.
+  Unless T is a ``RealmObject``, T can also be nullable (``RealmSet<T?>``).
```
Comment (mongomoe) [inline source/sdk/kotlin/realm-database/schemas/supported-types.txt:35]: Similar suggestion as above: https://github.com/mongodb/docs-realm/pull/2384/files#r1044837668

**Review by mongomoe — COMMENTED** [review]
Great job so far, but I think we're missing one type: `TypedRealmObject`. 
See:  https://www.mongodb.com/docs/realm-sdks/kotlin/latest/library-base/-realm%20-kotlin%20-s-d-k/io.realm.kotlin.types/-typed-realm-object/index.html

Can you ask the SDK team if this is a type that should be on the list of Supported types (I assume so because we can import it from io.realm.kotlin.types)? 

If you want to merge this PR while you wait for an answer on that, and create a separate PR + ticket for TypedRealmObject you can do that, or just hold onto this PR until we know the answer.

File: source/sdk/kotlin/realm-database/schemas/supported-types.txt [hunk 3]
```diff
@@ -17,15 +17,26 @@ Realm Database supports the following field data types:
...
 - ``RealmUUID``
+- ``BacklinksDelegate<T>``, which is used to define an inverse relationship between 
+  `RealmObjects <{+kotlin-local-prefix+}io.realm.kotlin.types/-realm-object/index.html>`__ using 
+  `backlinks <{+kotlin-local-prefix+}io.realm.kotlin.ext/backlinks.html>`__.
```
Comment (cmelchior) [inline source/sdk/kotlin/realm-database/schemas/supported-types.txt:39]:
```suggestion
- ``BacklinksDelegate<T>``, a `backlinks <{+kotlin-local-prefix+}io.realm.kotlin.ext/backlinks.html>` delegate used to define an inverse relationship between 
  `RealmObjects <{+kotlin-local-prefix+}io.realm.kotlin.types/-realm-object/index.html>`__. 
```

While technically correct, most people will not really understand it. The reference to the backlink docs are great, but I tried to reword a little to put more emphasis on it. Not 100% sure I got the formatting correct.

**Review by cmelchior — APPROVED** [review]
Looking good. Only a smaller nitpick around backlinks.

File: source/sdk/kotlin/realm-database/schemas/supported-types.txt (hunk 1 above)
Comment (cbullinger) [reply source/sdk/kotlin/realm-database/schemas/supported-types.txt:32]: Reworded

**Review by mongomoe — APPROVED** [review]
LGTM

**Pull request merged**